uast.cpp

```diff
@@ -0,0 +1,116 @@
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <assert.h>
+#include "uast.h"
+
+#define MAX_CHILDREN 100
```

Comment:
This constant appears twice, it should be shared in a header file.
Also, I don't think we need it as commented above.